<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Strict//EN">
<html>
<head><title>female_repl1_R1.qc.fq.gz FastQC Report</title>

<style type="text/css">

 @media screen {
  div.summary {
    width: 18em;
    position:fixed;
    top: 3em;
    margin:1em 0 0 1em;
  }
  
  div.main {
    display:block;
    position:absolute;
    overflow:auto;
    height:auto;
    width:auto;
    top:4.5em;
    bottom:2.3em;
    left:18em;
    right:0;
    border-left: 1px solid #CCC;
    padding:0 0 0 1em;
    background-color: white;
    z-index:1;
  }
  
  div.header {
    background-color: #EEE;
    border:0;
    margin:0;
    padding: 0.5em;
    font-size: 200%;
    font-weight: bold;
    position:fixed;
    width:100%;
    top:0;
    left:0;
    z-index:2;
  }

  div.footer {
    background-color: #EEE;
    border:0;
    margin:0;
	padding:0.5em;
    height: 1.3em;
	overflow:hidden;
    font-size: 100%;
    font-weight: bold;
    position:fixed;
    bottom:0;
    width:100%;
    z-index:2;
  }
  
  img.indented {
    margin-left: 3em;
  }
 }
 
 @media print {
	img {
		max-width:100% !important;
		page-break-inside: avoid;
	}
	h2, h3 {
		page-break-after: avoid;
	}
	div.header {
      background-color: #FFF;
    }
	
 }
 
 body {    
  font-family: sans-serif;   
  color: #000;   
  background-color: #FFF;
  border: 0;
  margin: 0;
  padding: 0;
  }
  
  div.header {
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 200%;
  font-weight: bold;
  width:100%;
  }    
  
  #header_title {
  display:inline-block;
  float:left;
  clear:left;
  }
  #header_filename {
  display:inline-block;
  float:right;
  clear:right;
  font-size: 50%;
  margin-right:2em;
  text-align: right;
  }

  div.header h3 {
  font-size: 50%;
  margin-bottom: 0;
  }
  
  div.summary ul {
  padding-left:0;
  list-style-type:none;
  }
  
  div.summary ul li img {
  margin-bottom:-0.5em;
  margin-top:0.5em;
  }
	  
  div.main {
  background-color: white;
  }
      
  div.module {
  padding-bottom:1.5em;
  padding-top:1.5em;
  }
	  
  div.footer {
  background-color: #EEE;
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 100%;
  font-weight: bold;
  width:100%;
  }


  a {
  color: #000080;
  }

  a:hover {
  color: #800000;
  }
      
  h2 {
  color: #800000;
  padding-bottom: 0;
  margin-bottom: 0;
  clear:left;
  }

  table { 
  margin-left: 3em;
  text-align: center;
  }
  
  th { 
  text-align: center;
  background-color: #000080;
  color: #FFF;
  padding: 0.4em;
  }      
  
  td { 
  font-family: monospace; 
  text-align: left;
  background-color: #EEE;
  color: #000;
  padding: 0.4em;
  }

  img {
  padding-top: 0;
  margin-top: 0;
  border-top: 0;
  }

  
  p {
  padding-top: 0;
  margin-top: 0;
  }
  
</style>

</head>
<body>
<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Wed 4 Mar 2015<br />
female_repl1_R1.qc.fq.gz
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>female_repl1_R1.qc.fq.gz</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>96615</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>25-100</td>
</tr>
<tr>
<td>%GC</td>
<td>49</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/warning.png" alt="[WARN]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/tick.png" alt="[OK]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GATCGGAAGAGCACACGTCTGAACTCCAGTCACCAGATCATCTCGTATGC</td>
<td>209</td>
<td>0.21632251720747295</td>
<td>TruSeq Adapter, Index 7 (100% over 50bp)</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>29975</td>
<td>3.229151</td>
<td>6.9134264</td>
<td>2</td>
</tr>
<tr>
<td>TTTTT</td>
<td>29170</td>
<td>3.124399</td>
<td>6.098522</td>
<td>2</td>
</tr>
<tr>
<td>TGCTG</td>
<td>25435</td>
<td>2.8241744</td>
<td>5.8074737</td>
<td>9</td>
</tr>
<tr>
<td>CTGCT</td>
<td>23940</td>
<td>2.6534665</td>
<td>5.560319</td>
<td>1</td>
</tr>
<tr>
<td>CTTCT</td>
<td>22995</td>
<td>2.516843</td>
<td>6.9162545</td>
<td>1</td>
</tr>
<tr>
<td>CAGCT</td>
<td>22005</td>
<td>2.4418027</td>
<td>5.514917</td>
<td>3</td>
</tr>
<tr>
<td>AAGAA</td>
<td>21375</td>
<td>2.3291748</td>
<td>5.6259036</td>
<td>2</td>
</tr>
<tr>
<td>CTCCT</td>
<td>20670</td>
<td>2.2869651</td>
<td>8.539177</td>
<td>1</td>
</tr>
<tr>
<td>CTTCA</td>
<td>20225</td>
<td>2.2162108</td>
<td>7.452785</td>
<td>1</td>
</tr>
<tr>
<td>CTCCA</td>
<td>18870</td>
<td>2.090214</td>
<td>6.999502</td>
<td>1</td>
</tr>
<tr>
<td>TCCAG</td>
<td>18195</td>
<td>2.0190232</td>
<td>5.059979</td>
<td>7</td>
</tr>
<tr>
<td>CTGGA</td>
<td>17770</td>
<td>1.9753634</td>
<td>7.8286953</td>
<td>1</td>
</tr>
<tr>
<td>CTGAA</td>
<td>17800</td>
<td>1.9561976</td>
<td>6.9975114</td>
<td>1</td>
</tr>
<tr>
<td>CTCTG</td>
<td>17080</td>
<td>1.8931162</td>
<td>5.560319</td>
<td>1</td>
</tr>
<tr>
<td>CTTCC</td>
<td>17035</td>
<td>1.8847824</td>
<td>5.550465</td>
<td>1</td>
</tr>
<tr>
<td>CTTTT</td>
<td>16870</td>
<td>1.8265917</td>
<td>6.8940954</td>
<td>1</td>
</tr>
<tr>
<td>CTTTG</td>
<td>16450</td>
<td>1.8036777</td>
<td>6.293858</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>14840</td>
<td>1.6686311</td>
<td>6.508556</td>
<td>1</td>
</tr>
<tr>
<td>CATCT</td>
<td>14955</td>
<td>1.6387361</td>
<td>5.258036</td>
<td>9</td>
</tr>
<tr>
<td>CTCAG</td>
<td>14650</td>
<td>1.6256493</td>
<td>8.457136</td>
<td>1</td>
</tr>
<tr>
<td>CTGAG</td>
<td>13900</td>
<td>1.5451633</td>
<td>5.2012568</td>
<td>1</td>
</tr>
<tr>
<td>GTTTT</td>
<td>13955</td>
<td>1.5136538</td>
<td>5.389034</td>
<td>1</td>
</tr>
<tr>
<td>CTGGC</td>
<td>13115</td>
<td>1.4720562</td>
<td>5.305901</td>
<td>1</td>
</tr>
<tr>
<td>GGATG</td>
<td>13150</td>
<td>1.4643862</td>
<td>6.2330475</td>
<td>4</td>
</tr>
<tr>
<td>CTTGG</td>
<td>12145</td>
<td>1.3485197</td>
<td>5.3559527</td>
<td>1</td>
</tr>
<tr>
<td>TGGAT</td>
<td>11550</td>
<td>1.2701213</td>
<td>5.4122267</td>
<td>3</td>
</tr>
<tr>
<td>ATGCA</td>
<td>10980</td>
<td>1.206688</td>
<td>6.3300886</td>
<td>6</td>
</tr>
<tr>
<td>CTCAT</td>
<td>10860</td>
<td>1.190015</td>
<td>5.497091</td>
<td>1</td>
</tr>
<tr>
<td>GATGC</td>
<td>10185</td>
<td>1.1321933</td>
<td>5.394413</td>
<td>5</td>
</tr>
<tr>
<td>CTCAA</td>
<td>10050</td>
<td>1.102525</td>
<td>5.2917504</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
</body></html>